baseplate/secrets/fetcher.py

```diff
@@ -272,12 +272,45 @@ def get_secret(self, secret_name):
         return payload["data"], ttl_to_time(payload["lease_duration"])
 
 
+def fetch_secrets(cfg, client_factory):
+    client = client_factory.get_client()
+    secrets = {}
+    soonest_expiration = client.token_expiration
+    for secret_name in cfg.secrets:
+        secrets[secret_name], expiration = client.get_secret(secret_name)
+        soonest_expiration = min(soonest_expiration, expiration)
+
+    with open(cfg.output.path + ".tmp", "w") as f:
+        os.fchown(f.fileno(), cfg.output.owner, cfg.output.group)
+        os.fchmod(f.fileno(), cfg.output.mode)
+
+        json.dump({
+            "secrets": secrets,
+            "vault": {
+                "token": client.token,
+                "url": cfg.vault.url,
+            },
+
+            # this is here to allow an upgrade path. the fetcher should
+            # be upgraded first followed by the application workers.
+            "vault_token": client.token,
+        }, f, indent=2, sort_keys=True)
+
+    # swap out the file contents atomically
+    os.rename(cfg.output.path + ".tmp", cfg.output.path)
+    return soonest_expiration
+
+
 def main():
     arg_parser = argparse.ArgumentParser()
     arg_parser.add_argument("config_file", type=argparse.FileType("r"),
         help="path to a configuration file")
     arg_parser.add_argument("--debug", default=False, action="store_true",
         help="enable debug logging")
+    arg_parser.add_argument("--mode", type=str, default="daemon",
```

Comment:
How would we feel about a single `--run-once` flag instead of mode? I feel like the latter may not be immediately evident at a glance, and could be confused with other things.

Reply:
noice

Reply:
beat me to it, @spladug

Reply:
:highfive: